### PR TITLE
refactor(unlock-assets): Generate typescript components

### PR DESCRIPTION
**Pull request by jordang7**
# Description

Modified unlock-assets package to have svgr generate typescript components
- Removed js files in `build/svg-components` and replaced with typescript
- Added `interfaces` variable to `template.js `to allow for typescript and `--typescript` flag added to` svg-2-components` script

# Issues

#8851 

Fixes #
Refs #

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: packages/unlock-assets/src/SvgComponents.js [hunk 1]
```diff
@@ -1,69 +1,69 @@
...
+import Unlock from '../build/svg-component/Unlock.tsx'
+import UnlockWordMark from '../build/svg-component/UnlockWordMark.tsx'
+import Upload from '../build/svg-component/Upload.tsx'
+import Wallet from '../build/svg-component/Wallet.tsx'
+import WalletConnect from '../build/svg-component/WalletConnect.tsx'
+import Webflow from '../build/svg-component/Webflow.tsx'
+import Withdraw from '../build/svg-component/Withdraw.tsx'
+import Wordpress from '../build/svg-component/Wordpress.tsx'
+import UnlockMonogram from '../build/svg-component/UnlockMonogram.tsx'
+import RocketLaunch from '../build/svg-component/RocketLaunch.tsx'
```
[the remark this answers is not in the document]
Comment (jordang7) [reply packages/unlock-assets/src/SvgComponents.js]: Just pushed a commit that should resolve this

File: packages/unlock-assets/dist/index.js [hunk 2]
```diff
@@ -40,349 +40,232 @@ function createCommonjsModule(fn, module) {
 	return module = { exports: {} }, fn(module, module.exports), module.exports;
 }
```
[the remark this answers is not in the document]
Comment (jordang7) [reply packages/unlock-assets/dist/index.js]: Removed the build files, will check the CI and local scripts to make sure they are building this package

**Comment by jordang7** [comment]
Yeah looking into it now. Everything is working for me locally after adding unlock-assets to the build script, but it seems like the Unit tests are failing here.

**Comment by jordang7** [comment]
Removing the build files from the repo is causing the tests to fail.

I've added `unlock-assets` to the build script, but haven't been able to find the solution. Is there anywhere else it needs to be added so that the package is built and available during testing?

**Comment by jordang7** [comment]
Okay sounds good, just created another PR with just the typescript changes here https://github.com/unlock-protocol/unlock/pull/9515

**Comment by searchableguy** [comment]
I'm closing this PR. :) I will change the CI. Thanks for the effort.

**Pull request closed without merging**